share/completions/ffmpeg.fish

```diff
@@ -0,0 +1,102 @@
+complete -c ffmpeg -s i -d "Specify input file"
+
+# Print help / information / capabilities
+complete -c ffmpeg -s L -d "Show license"
+complete -x -c ffmpeg -s h -s "?" -o help -l help -a "
+long
+full
+(ffmpeg -loglevel quiet -decoders | grep -v -e '=' -e ':\$' -e '^\s-' | sed 's/[\t ]/\t/g' | cut -f 3 | sed 's/^/decoder=/g')
```

Comment:
Also this seems like prime use of a for-loop or even function:

```fish
for thing in encoder demuxer muxer filter
    printf '%s\n' $thing=(ffmpeg -loglevel quiet -{$thing}s | ...)
end
```

Reply:
And instead of using `cut` based on whitespace, which has to use a different field just because this does a positional "ED" thing, use `string trim` to trim leading whitespace. That way it won't break if e.g. ffmpeg decides to actually display "ED" for some of these, or to start displaying it as ".D" (like the legend says!)

Reply:
I.e. I'd roughly do this as

```fish
for thing in encoder demuxer muxer filter
    printf '%s\n' $thing=(ffmpeg -loglevel quiet -"$thing"s | 
                          string trim |
                          string match -rv '=|:\$|^\s-' | 
                          string replace -rf '\S+\s+(\S+)\s+(\S+)' '$1\t$2')
end
```